### PR TITLE
perf: Added prisma optimize support with flag

### LINKED ISSUE #15795: perf: enable prisma optimize
### Is your proposal related to a problem?

(Write your answer here.)
We can enable prisma to enhance an existing application or explore its capabilities through a demo.

### Describe the solution you'd like

(Describe your proposed solution here.)

### Describe alternatives you've considered

(Write your answer here.)

### Additional context

(Write your answer here.)

### Requirement/Document
It can be easily enabled with referring this [https://www.prisma.io/blog/prisma-optimize-early-access](https://www.prisma.io/blog/prisma-optimize-early-access)

(Share it here.)

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

### PR DESCRIPTION
**Pull request by harshsbhat**
## What does this PR do?

- Fixes #15795 (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://www.loom.com/share/30ffb9c4cd5843a08c0ec0e5a35fdef6?sid=e428ac8c-8799-442a-bd31-f84cf410c102

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
Yes! As requested in my previous PR. env variable should  have ENABLE_OPTIMIZE_EXTENSION="true" to run prisma optimize. Else Prisma Optimize won't start.

### CODE REVIEW

**Comment by harshsbhat** [comment]
@keithwillcode @PeerRich can someone review this?

File: packages/prisma/index.ts [hunk 1]
```diff
@@ -25,7 +26,8 @@ export const customPrisma = (options?: Prisma.PrismaClientOptions) =>
     .$extends(usageTrackingExtention())
     .$extends(excludePendingPaymentsExtension())
     .$extends(bookingIdempotencyKeyExtension())
-    .$extends(withAccelerate());
+    .$extends(withAccelerate())
+    .$extends(withOptimize());
```
Comment (mhessdev) [inline packages/prisma/index.ts:30]: You will need to pass in the API key here  `withOptimize({ token: process.env.OPTIMIZE_API_KEY })`

**Review by mhessdev — CHANGES_REQUESTED** [review]

File: packages/prisma/index.ts (hunk 1 above)
Comment (harshsbhat) [reply packages/prisma/index.ts:30]: Last time I checked the withOptimize didn't require an API key. Can you send me the documentation that you are referring to?

Comment (apolanc) [reply packages/prisma/index.ts:30]: Hi @harshsbhat, we launched a new version of Optimize recently. You can read more in the [docs](https://www.prisma.io/docs/optimize/getting-started)!

File: package.json [hunk 2]
```diff
@@ -83,6 +83,7 @@
   "devDependencies": {
     "@changesets/cli": "^2.26.1",
     "@playwright/test": "^1.45.3",
+    "@prisma/extension-optimize": "0.0.0-dev.202407222340",
```
[the remark this answers is not in the document]
Comment (apolanc) [reply package.json:86]: Current version is now `1.0.1` -> https://www.npmjs.com/package/@prisma/extension-optimize

**Comment by keithwillcode** [comment]
Have been testing this and can't get anything to show in the dashboard. Will revisit soon.

**Comment by harshsbhat** [comment]
@keithwillcode . Its because when I made the PR they didn't have an API key support. It was in beta mode. They have officially launched it now. Give me a few days I will get this done. A little busy right now with some other work

**Comment by harshsbhat** [comment]
> > @keithwillcode . Its because when I made the PR they didn't have an API key support. It was in beta mode. They have officially launched it now. Give me a few days I will get this done. A little busy right now with some other work
> 
> 
> 
> I went ahead and made the changes for the API key support. It might be the ordering in which we wire things up. Just don’t have too much more time to dedicate to fixing it at the moment. 

Yeah, I am kind of busy for a few days. You dont mind me taking on this next week right?

**Comment by keithwillcode** [comment]
> > > @keithwillcode . Its because when I made the PR they didn't have an API key support. It was in beta mode. They have officially launched it now. Give me a few days I will get this done. A little busy right now with some other work
> > 
> > 
> > I went ahead and made the changes for the API key support. It might be the ordering in which we wire things up. Just don’t have too much more time to dedicate to fixing it at the moment.
> 
> Yeah, I am kind of busy for a few days. You dont mind me taking on this next week right?

Not at all. 🙏🏼

**Comment by keithwillcode** [comment]
Closing for now until work is picked back up on this.

**Pull request closed without merging**